### PR TITLE
Expand permissions system

### PR DESCRIPTION
**Pull request by joaoburatto**
# Summary

Expands and cleans up the permissions systems to facilitate changes, adds the missing change permission and save permissions functionality.

Creates a permission settings file to define if a server/host should get the owner permission whenever they join the game.

Adds the functionality to change a user's permission, and the functionality to save the permissions back to the permissions file, this can be used if the permissions are changed runtime, ex. when an admin gives a non admin the admin permission.

#### PR checklist
- [ ] The game builds properly without errors.

- [ ] No unrelated changes are present.

- [ ] No "trash" files are committed.

- [ ] Relevant code is documented.

- [ ] Update the related GitBook document, or create a new one if needed.

# Testing

This can be tested by running the game as the host, with the **AddServerOwnerPermissionToServerHost**  setting ticked under "ProjectSettings/SS3D/Server/PermissionSettings", this should change the host permission to "Server Owner" whenever the server is started. There should be logs for this.

#### Networking checklist

- [ ] Works from host in host mode.

- [ ] Works from server in server mode.

- [ ] Works on server in client mode.

- [ ] Works and is syncronized across different clients.

- [ ] Is persistent.

# Changes

Many changes are related to import settings, the permissions system was moved to a separate assembly as it is very independent and that makes things more modular.

### CODE REVIEW

File: Assets/Scripts/SS3D/Permissions/PermissionSettings.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Coimbra;
+using UnityEngine;
+
+namespace SS3D.Permissions
+{
+    [ProjectSettings("SS3D/Server")]
+    public class PermissionSettings : ScriptableSettings
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Permissions/PermissionSettings.cs:10]: Add some doc

File: Assets/Scripts/SS3D/Permissions/PermissionSystem.cs [hunk 2]
```diff
@@ -87,15 +88,30 @@ public bool TryGetUserRole(string ckey, out ServerRoleTypes userPermission)
...
+            ServerRoleTypes previousRole = _userPermissions.TryGetValue(ckey, out ServerRoleTypes permission) ? permission : ServerRoleTypes.None;
+
+            Log.Information(this, $"Updating user {ckey} role from {previousRole} to {role}");
+
+            _userPermissions[ckey] = role;
+
+            SaveUserPermissions();
+        }
+
+        public void SaveUserPermissions()
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Permissions/PermissionSystem.cs:108]: add some doc

**Review by stilnat — CHANGES_REQUESTED** [review]
Just a bit of doc to add, and explain a bit more how this works from the user perspective in the PR description. Do I choose to register permission ? Is is automatic ? Can you do it on build or only in editor ?

File: Assets/Scripts/SS3D/Permissions/PermissionSettings.cs (hunk 1 above)
Comment (joaoburatto) [reply Assets/Scripts/SS3D/Permissions/PermissionSettings.cs:10]: Added

File: Assets/Scripts/SS3D/Permissions/PermissionSystem.cs (hunk 2 above)
Comment (joaoburatto) [reply Assets/Scripts/SS3D/Permissions/PermissionSystem.cs:108]: Added

**Review by stilnat — CHANGES_REQUESTED** [review]
Please add a detailed description of the PR

**Comment by joaoburatto** [comment]
@stilnat added docs and description

**Review by stilnat — CHANGES_REQUESTED** [review]
fix build error

**Review by stilnat — APPROVED** [review]
works as expected, good job

**Pull request merged**